gdlsc_test.go

```diff
@@ -0,0 +1,92 @@
+package main
```

Comment:
This is good, just few things:

1) Use this package for assertions: https://github.com/stretchr/testify
2) It also provides some nice generic equality functions so you won't need (n *Node) equal(...)
3) Add some more test cases:
- Another level of depth
- Node with license with a child with a license too